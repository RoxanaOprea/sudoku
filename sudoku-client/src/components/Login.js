import React from "react";
import TextField from "@material-ui/core/TextField";
import { withStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";

const styles = theme => ({
  container: {
    display: "flex",
    flexWrap: "wrap",
    marginTop: "150px",
    width: "300px",
    margin: "auto"
  },
  textField: {
    margin: "auto",
    marginBottom: "15px"
  },
  button: {
    margin: "110px"
  },
  input: {
    display: "none"
  },
  root: {
    flexGrow: 1
  },
  para: {
    margin: "auto",
    paddingBottom: "10px"
  }
});

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: null,
      password: null
    };
  }

  handleChange(event, inputName) {
    this.setState({ [inputName]: event.target.value });
  }

  handleSubmit = (event, email, password) => {
    event.preventDefault();

    fetch("http://localhost:3001/login", {
      method: "POST",
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password
      }),
      headers: {
        "Content-Type": "application/json"
      }
    })
      .then(response => response.json())
      .then(user => console.log(`Login ${user}`))
      .catch(err => console.log(err));
      
  };

  render() {
    const { classes } = this.props;
    return (
      <div>
        <MuiThemeProvider>
          <div>
            <AppBar title="Login">
              <Toolbar>
                <Typography variant="h6" color="inherit">
                  Login Page
                </Typography>
              </Toolbar>
            </AppBar>
            <form
              className={classes.container}
              noValidate
              autoComplete="off"
              onSubmit={this.handleSubmit}
            >
              <TextField
                id="outlined-email-input"
                label="Email"
                className={classes.textField}
                type="email"
                name="email"
                autoComplete="email"
                margin="normal"
                variant="outlined"
                onChange={event => this.handleChange(event, "email")}
              />
              <TextField
                id="outlined-password-input"
                label="Password"
                className={classes.textField}
                type="password"
                autoComplete="current-password"
                margin="normal"
                variant="outlined"
                onChange={event => this.handleChange(event, "password")}
              />
              <div className={classes.button}>
                <Button type="Submit" variant="outlined" color="primary">
                  Login
                </Button>
              </div>
            </form>
            <p className={classes.para}>Not registered yet? Register Now!</p>
            <Link to="/register">Register</Link>
          </div>
        </MuiThemeProvider>
      </div>
    );
  }
}

Login.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Login);
