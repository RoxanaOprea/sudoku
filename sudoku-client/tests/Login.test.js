import React from "react";
import { shallow } from "enzyme";
import Login from "../src/components/Login";
//import { spy } from "sinon";
//import toJson from "enzyme-to-json";
// import Link from "react-router-dom";
// import { render } from "react-dom";

jest.mock("react-dom");

let component = null;

describe("Login component", () => {
  beforeEach(() => {
    component = shallow(<Login />);
  });

  it("renders correctly", () => {
    expect(component).toMatchSnapshot();
    expect(component.dive()).toMatchSnapshot();
  });

  it("should have a `form` element", () => {
    expect(component.dive().find("form").length).toBe(1);
  });

  it("renders a text field for email", () => {
    expect(component.dive().find("#login-email-input").length).toEqual(1);
  });

  it("renders a text field for password", () => {
    expect(component.dive().find("#login-password-input").length).toEqual(1);
  });

  it("renders a button", () => {
    expect(component.dive().find("#login-button").length).toEqual(1);
  });

  it("renders a link", () => {
    expect(component.dive().find("#register-page").length).toEqual(1);
  });
});


describe("Login component actions", () => {
   beforeEach(() => {
     component = shallow(<Login />).dive();
  });

  it("should respond to change event and change the state for email field", () => {
    const emailInput = component.find("#login-email-input");
    emailInput.first().simulate("change", {
      target: { name: "email", value: "test@test.com" }
    });

    expect(component.instance().state.email).toEqual("test@test.com");
  });

  it("should respond to change event and change the state for password field", () => {
    const passwordInput = component.find("#login-password-input");
    passwordInput.first().simulate("change", {
      target: { name: "password", value: "test" }
    });

    expect(component.instance().state.password).toEqual("test");
  });

 });

