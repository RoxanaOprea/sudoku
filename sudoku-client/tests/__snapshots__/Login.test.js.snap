// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Login) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "Login-button-3",
        "container": "Login-container-1",
        "error": "Login-error-7",
        "input": "Login-input-4",
        "para": "Login-para-6",
        "root": "Login-root-5",
        "textField": "Login-textField-2",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "Login-button-3",
          "container": "Login-container-1",
          "error": "Login-error-7",
          "input": "Login-input-4",
          "para": "Login-para-6",
          "root": "Login-root-5",
          "textField": "Login-textField-2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Login component renders correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    classes={
      Object {
        "button": "Login-button-3",
        "container": "Login-container-1",
        "error": "Login-error-7",
        "input": "Login-input-4",
        "para": "Login-para-6",
        "root": "Login-root-5",
        "textField": "Login-textField-2",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <WithStyles(AppBar)
          title="Login"
        >
          <WithStyles(Toolbar)>
            <WithStyles(Typography)
              color="inherit"
              variant="h6"
            >
              Login Page
            </WithStyles(Typography)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>
        <form
          autoComplete="off"
          className="Login-container-1"
          noValidate={true}
          onSubmit={[Function]}
        >
          <TextField
            autoComplete="email"
            className="Login-textField-2"
            id="login-email-input"
            label="Email"
            margin="normal"
            name="email"
            onChange={[Function]}
            required={false}
            select={false}
            type="email"
            variant="outlined"
          />
          <TextField
            autoComplete="current-password"
            className="Login-textField-2"
            id="login-password-input"
            label="Password"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            type="password"
            variant="outlined"
          />
          <div
            className="Login-button-3"
          >
            <WithStyles(Button)
              color="primary"
              id="login-button"
              type="Submit"
              variant="outlined"
            >
              Login
            </WithStyles(Button)>
          </div>
        </form>
        <p
          className="Login-para-6"
        >
          Not registered yet? Register Now!
        </p>
        <Link
          id="register-page"
          replace={false}
          title="Register"
          to="/register"
        >
          Register
        </Link>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(AppBar)
            title="Login"
          >
            <WithStyles(Toolbar)>
              <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Login Page
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>,
          <form
            autoComplete="off"
            className="Login-container-1"
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextField
              autoComplete="email"
              className="Login-textField-2"
              id="login-email-input"
              label="Email"
              margin="normal"
              name="email"
              onChange={[Function]}
              required={false}
              select={false}
              type="email"
              variant="outlined"
            />
            <TextField
              autoComplete="current-password"
              className="Login-textField-2"
              id="login-password-input"
              label="Password"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              variant="outlined"
            />
            <div
              className="Login-button-3"
            >
              <WithStyles(Button)
                color="primary"
                id="login-button"
                type="Submit"
                variant="outlined"
              >
                Login
              </WithStyles(Button)>
            </div>
          </form>,
          <p
            className="Login-para-6"
          >
            Not registered yet? Register Now!
          </p>,
          <Link
            id="register-page"
            replace={false}
            title="Register"
            to="/register"
          >
            Register
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Login Page
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>,
            "title": "Login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Login Page
              </WithStyles(Typography)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login Page",
                "color": "inherit",
                "variant": "h6",
              },
              "ref": null,
              "rendered": "Login Page",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "children": Array [
              null,
              <TextField
                autoComplete="email"
                className="Login-textField-2"
                id="login-email-input"
                label="Email"
                margin="normal"
                name="email"
                onChange={[Function]}
                required={false}
                select={false}
                type="email"
                variant="outlined"
              />,
              <TextField
                autoComplete="current-password"
                className="Login-textField-2"
                id="login-password-input"
                label="Password"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="outlined"
              />,
              <div
                className="Login-button-3"
              >
                <WithStyles(Button)
                  color="primary"
                  id="login-button"
                  type="Submit"
                  variant="outlined"
                >
                  Login
                </WithStyles(Button)>
              </div>,
            ],
            "className": "Login-container-1",
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoComplete": "email",
                "className": "Login-textField-2",
                "id": "login-email-input",
                "label": "Email",
                "margin": "normal",
                "name": "email",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "email",
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoComplete": "current-password",
                "className": "Login-textField-2",
                "id": "login-password-input",
                "label": "Password",
                "margin": "normal",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <WithStyles(Button)
                  color="primary"
                  id="login-button"
                  type="Submit"
                  variant="outlined"
                >
                  Login
                </WithStyles(Button)>,
                "className": "Login-button-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "color": "primary",
                  "id": "login-button",
                  "type": "Submit",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Not registered yet? Register Now!",
            "className": "Login-para-6",
          },
          "ref": null,
          "rendered": "Not registered yet? Register Now!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Register",
            "className": undefined,
            "id": "register-page",
            "replace": false,
            "title": "Register",
            "to": "/register",
          },
          "ref": null,
          "rendered": "Register",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <WithStyles(AppBar)
            title="Login"
          >
            <WithStyles(Toolbar)>
              <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Login Page
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>
          <form
            autoComplete="off"
            className="Login-container-1"
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextField
              autoComplete="email"
              className="Login-textField-2"
              id="login-email-input"
              label="Email"
              margin="normal"
              name="email"
              onChange={[Function]}
              required={false}
              select={false}
              type="email"
              variant="outlined"
            />
            <TextField
              autoComplete="current-password"
              className="Login-textField-2"
              id="login-password-input"
              label="Password"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              variant="outlined"
            />
            <div
              className="Login-button-3"
            >
              <WithStyles(Button)
                color="primary"
                id="login-button"
                type="Submit"
                variant="outlined"
              >
                Login
              </WithStyles(Button)>
            </div>
          </form>
          <p
            className="Login-para-6"
          >
            Not registered yet? Register Now!
          </p>
          <Link
            id="register-page"
            replace={false}
            title="Register"
            to="/register"
          >
            Register
          </Link>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(AppBar)
              title="Login"
            >
              <WithStyles(Toolbar)>
                <WithStyles(Typography)
                  color="inherit"
                  variant="h6"
                >
                  Login Page
                </WithStyles(Typography)>
              </WithStyles(Toolbar)>
            </WithStyles(AppBar)>,
            <form
              autoComplete="off"
              className="Login-container-1"
              noValidate={true}
              onSubmit={[Function]}
            >
              <TextField
                autoComplete="email"
                className="Login-textField-2"
                id="login-email-input"
                label="Email"
                margin="normal"
                name="email"
                onChange={[Function]}
                required={false}
                select={false}
                type="email"
                variant="outlined"
              />
              <TextField
                autoComplete="current-password"
                className="Login-textField-2"
                id="login-password-input"
                label="Password"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="outlined"
              />
              <div
                className="Login-button-3"
              >
                <WithStyles(Button)
                  color="primary"
                  id="login-button"
                  type="Submit"
                  variant="outlined"
                >
                  Login
                </WithStyles(Button)>
              </div>
            </form>,
            <p
              className="Login-para-6"
            >
              Not registered yet? Register Now!
            </p>,
            <Link
              id="register-page"
              replace={false}
              title="Register"
              to="/register"
            >
              Register
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Toolbar)>
                <WithStyles(Typography)
                  color="inherit"
                  variant="h6"
                >
                  Login Page
                </WithStyles(Typography)>
              </WithStyles(Toolbar)>,
              "title": "Login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)
                  color="inherit"
                  variant="h6"
                >
                  Login Page
                </WithStyles(Typography)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login Page",
                  "color": "inherit",
                  "variant": "h6",
                },
                "ref": null,
                "rendered": "Login Page",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "children": Array [
                null,
                <TextField
                  autoComplete="email"
                  className="Login-textField-2"
                  id="login-email-input"
                  label="Email"
                  margin="normal"
                  name="email"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="email"
                  variant="outlined"
                />,
                <TextField
                  autoComplete="current-password"
                  className="Login-textField-2"
                  id="login-password-input"
                  label="Password"
                  margin="normal"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="password"
                  variant="outlined"
                />,
                <div
                  className="Login-button-3"
                >
                  <WithStyles(Button)
                    color="primary"
                    id="login-button"
                    type="Submit"
                    variant="outlined"
                  >
                    Login
                  </WithStyles(Button)>
                </div>,
              ],
              "className": "Login-container-1",
              "noValidate": true,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoComplete": "email",
                  "className": "Login-textField-2",
                  "id": "login-email-input",
                  "label": "Email",
                  "margin": "normal",
                  "name": "email",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "email",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoComplete": "current-password",
                  "className": "Login-textField-2",
                  "id": "login-password-input",
                  "label": "Password",
                  "margin": "normal",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "password",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <WithStyles(Button)
                    color="primary"
                    id="login-button"
                    type="Submit"
                    variant="outlined"
                  >
                    Login
                  </WithStyles(Button)>,
                  "className": "Login-button-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "color": "primary",
                    "id": "login-button",
                    "type": "Submit",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Not registered yet? Register Now!",
              "className": "Login-para-6",
            },
            "ref": null,
            "rendered": "Not registered yet? Register Now!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Register",
              "className": undefined,
              "id": "register-page",
              "replace": false,
              "title": "Register",
              "to": "/register",
            },
            "ref": null,
            "rendered": "Register",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
